File: pptx-service/renderers/question.py
from builder import add_bg, add_textbox
from theme import ThemeColors


def render(slide, data: dict, t: ThemeColors) -> None:
    add_bg(slide, t.bg)
    question = data.get("question", "")
    hints = data.get("hints", [])
    invitation = data.get("invitation", "")

    add_textbox(slide, 10.0, 0.5, 3.0, 6.0, "?",
                t.font_display, 360, t.accent, bold=True)

    add_textbox(slide, 0.83, 1.5, 9.5, 4.0, question,
                t.font_display, 56, t.text, bold=True, wrap=True)

    y = 5.8
    for hint in (hints or [])[:3]:
        add_textbox(slide, 0.83, y, 9.0, 0.5, f"· {hint}",
                    t.font_body, 22, t.muted)
        y += 0.45

    if invitation:
        add_textbox(slide, 0.83, 6.8, 9.0, 0.5, invitation,
                    t.font_body, 20, t.muted, italic=True)

File: lib/pptx/slides/question.ts
import type { QuestionSlide } from '@/lib/types';
import {
  py, SLIDE_W,
  setBackground, addTextBox, addPageNumber,
  type PptxTheme,
} from '../helpers';
import type PptxGenJS from 'pptxgenjs';
type PSlide = ReturnType<InstanceType<typeof PptxGenJS>['addSlide']>;

export function render(slide: PSlide, data: QuestionSlide, t: PptxTheme, n: number, total: number): void {
  setBackground(slide, '#' + t.bg);

  addTextBox(slide, data.question, {
    x: t.paddingIn, y: py(300), w: SLIDE_W - 2 * t.paddingIn,
    fontSize: Math.round(t.heroSize * 0.7), color: '#' + t.text,
    fontFace: t.fontDisplay, bold: true, align: 'center', wrap: true,
  });

  if (data.hints && data.hints.length > 0) {
    addTextBox(slide, data.hints.map(h => `• ${h}`).join('\n'), {
      x: t.paddingIn, y: py(620), w: SLIDE_W - 2 * t.paddingIn,
      fontSize: t.bodySize, color: '#' + t.muted, fontFace: t.fontBody, align: 'center', wrap: true,
    });
  }

  if (data.invitation) {
    addTextBox(slide, data.invitation, {
      x: t.paddingIn, y: py(850), w: SLIDE_W - 2 * t.paddingIn,
      fontSize: t.captionSize, color: '#' + t.accent, fontFace: t.fontBody, italic: true, align: 'center',
    });
  }

  addPageNumber(slide, n, total, t);
}
